client/app/components/FavoritesControl.jsx

```diff
@@ -0,0 +1,80 @@
+import { isFunction } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { $rootScope } from '@/services/ng';
+
+function toggleItem(event, item, callback) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  const action = item.is_favorite ? item.$unfavorite.bind(item) : item.$favorite.bind(item);
+  const savedIsFavorite = item.is_favorite;
+
+  action().then(() => {
+    item.is_favorite = !savedIsFavorite;
+    $rootScope.$broadcast('reloadFavorites');
+    if (isFunction(callback)) {
```

Comment:
Since there's a default value to `onChange` isn't this check redundant?

Reply:
@ranbena Fixed